### PR TITLE
fix: invite flow rework

### PR DESCRIPTION
**Pull request by JivusAyrus**
- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

FIXES ENG-4561
FIXES ENG-4557

### CODE REVIEW

File: controlplane/package.json [hunk 1]
```diff
@@ -64,6 +65,7 @@
     "pg-boss": "^9.0.3",
     "pino": "^8.16.1",
     "postgres": "^3.4.3",
+    "postmark": "^3.11.0",
```
Comment (StarpTech) [inline controlplane/package.json:68]: This must be vendor neutral. Email should be sent over SMTP. You can use https://github.com/nodemailer/nodemailer/

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: controlplane/package.json (hunk 1 above)
Comment (JivusAyrus) [reply controlplane/package.json]: Done

File: controlplane/src/core/services/Nodemailer.ts [hunk 2]
```diff
@@ -8,11 +8,19 @@ interface OrganizationInviteBody {
   inviteLink: string;
 }
 
-export default class Postmark {
-  client: postmark.ServerClient;
+export default class Nodemailer {
```
Comment (StarpTech) [inline controlplane/src/core/services/Nodemailer.ts:11]: Mailer?

File: controlplane/src/core/services/Nodemailer.ts [hunk 3]
```diff
@@ -8,11 +8,19 @@ interface OrganizationInviteBody {
...
+  client: Transporter;
 
   constructor(serverToken: string) {
-    this.client = new postmark.ServerClient(serverToken);
+    this.client = createTransport({
+      host: 'smtp.postmarkapp.com',
+      port: 587,
+      secure: false,
+      auth: {
+        user: serverToken,
```
Comment (StarpTech) [inline controlplane/src/core/services/Nodemailer.ts:20]: This should be configurable. On-premise some might really use a username / password combination. We will just use the token.

File: controlplane/src/resources/organizationInvite.html
Comment (StarpTech) [inline controlplane/src/resources/organizationInvite.html:1]: Can we call the directory `templates/email` or similiar?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: controlplane/src/core/services/Nodemailer.ts (hunk 3 above)
Comment (JivusAyrus) [reply controlplane/src/core/services/Nodemailer.ts]: Done

File: controlplane/src/resources/organizationInvite.html
Comment (JivusAyrus) [reply controlplane/src/resources/organizationInvite.html:1]: Done

File: controlplane/src/core/services/Nodemailer.ts (hunk 2 above)
Comment (JivusAyrus) [reply controlplane/src/core/services/Nodemailer.ts]: Done

File: controlplane/src/core/repositories/OrganizationInvitationRepository.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      name: org.name,
+      slug: org.slug,
+      creatorUserId: org.creatorUserId,
+      createdAt: org.createdAt.toISOString(),
+    }));
+
+    return userInvitations;
+  }
+
+  public async getPendingOrganizationInvitation(input: {
```
Comment (StarpTech) [inline controlplane/src/core/repositories/OrganizationInvitationRepository.ts:62]: Why just a single one?

File: controlplane/src/db/schema.ts [hunk 5]
```diff
@@ -852,3 +851,17 @@ export const organizationLimits = pgTable('organization_limits', {
...
+  id: uuid('id').notNull().primaryKey().defaultRandom(),
+  organizationId: uuid('organization_id')
+    .notNull()
+    .references(() => organizations.id, {
+      onDelete: 'cascade',
+    }),
+  userId: uuid('user_id')
+    .notNull()
+    .references(() => users.id, { onDelete: 'cascade' }),
+  acceptedInvite: boolean('accepted_invite').default(false),
```
Comment (StarpTech) [inline controlplane/src/db/schema.ts:865]: Maybe just `accepted`? The table is already scoped to invitations.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: controlplane/src/core/repositories/OrganizationInvitationRepository.ts (hunk 4 above)
Comment (JivusAyrus) [reply controlplane/src/core/repositories/OrganizationInvitationRepository.ts:62]: To check if the invitation already exists

File: controlplane/src/db/schema.ts (hunk 5 above)
Comment (JivusAyrus) [reply controlplane/src/db/schema.ts]: Done

File: controlplane/src/core/repositories/OrganizationInvitationRepository.ts (hunk 4 above)
Comment (StarpTech) [reply controlplane/src/core/repositories/OrganizationInvitationRepository.ts:62]: thx

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**